test/distributed/fsdp/test_fsdp_ignored_modules.py

```diff
@@ -211,40 +221,70 @@ def test_diff_ignored_modules_across_ranks(
                 all ignored modules (representing a superset of the children's
                 ignored modules) to the root FSDP instance.
         """
+        self.run_subtests(
+            {"pass_ignored_modules_to_root": [False, True], "ignore_modules": [True, False]},
+            self._test_diff_ignored_modules_across_ranks,
+        )
+
+    def _test_diff_ignored_modules_across_ranks(self, pass_ignored_modules_to_root: bool, ignore_modules: bool):
         # To exercise different `FlatParameter` enumerations across ranks,
         # we wrap `layer3` with FSDP, where `layer3` is registered as a module
         # after `layer1`, which has the variable number of ignored modules
         model = ModelWithIgnoredModules(num_ignored=self.rank + 1).cuda()
         layer1_ignored_modules = [
             m for m in model.layer1.modules() if isinstance(m, IgnoredModule)
         ]
-        model.layer1 = FSDP(model.layer1, ignored_modules=layer1_ignored_modules)
+        if ignore_modules:
+            model.layer1 = FSDP(model.layer1, ignored_modules=layer1_ignored_modules)
+        else:
+            model.layer1 = FSDP(model.layer1, ignored_parameters=set(
+                p for m in layer1_ignored_modules for p in m.parameters()))
         model.layer3 = FSDP(model.layer3)
         model_ignored_modules = (
             [m for m in model.modules() if isinstance(m, IgnoredModule)]
             if pass_ignored_modules_to_root
             else []
         )
-        wrapped_model = FSDP(model, ignored_modules=model_ignored_modules)
+        if ignore_modules:
+            wrapped_model = FSDP(model, ignored_modules=model_ignored_modules)
+        else:
+            wrapped_model = FSDP(model, ignored_parameters=set(
+                p for m in model_ignored_modules for p in m.parameters()))
         optim = torch.optim.Adam(wrapped_model.parameters(), lr=1e-3)
         self._train_model(wrapped_model, optim, 3)
 
     @skip_if_lt_x_gpu(2)
-    def test_ignored_modules_not_under_wrapped_root(self):
+    @parametrize("ignore_modules", [True, False])
+    def test_ignored_modules_not_under_wrapped_root(self, ignore_modules: bool):
         model = Model().cuda()
         ignored_modules = list(model.layer1.children())[1:]
```

Comment:
nit: To avoid duplication
```
ignore_kwargs = (
    {"ignored_modules": ignored_modules}
    if ignore_modules
    else {"ignored_parameters": set(p for m in ignored_modules for p in m.parameters())}
)
model.layer1 = FSDP(
    model.layer1,
    # sharding_strategy shouldn't matter here.
    sharding_strategy=ShardingStrategy.SHARD_GRAD_OP,
    **ignore_kwargs,
)
model.layer3 = FSDP(
    model.layer3,
    # the ignored modules/parameters contains submodule under model.layer1, which
    # is out of the local root model.layer3.
    **ignore_kwargs,
)
```

Reply:
nit: We can do something something similar for the other unit tests, but it does not need to be changed for this PR. The idea is that we can push the `if ignore_modules` / `else` branch to the tightest scope possible to avoid duplication. That way, other kwargs like `use_orig_params`, etc. do not need to be repeated. Otherwise, if we ever add a new kwargs, we will need to add it to two FSDP constructor calls per unit test.